x-pack/plugin/core/src/main/resources/logs-json-pipeline.json

```diff
@@ -0,0 +1,48 @@
+{
```

Comment:
We need to discuss the naming of this pipeline. Because at the moment it would conflict with a pipeline for the `dataset: json` and `namespace: pipeline`. We need to come up with a non conflicting naming convention for these global asset. Ideally the component templates and pipelines follow the same logic. @kpollich @joshdover You might have ideas here too.

Reply:
What's the point of this being a separate pipeline at all? Are we going to be reusing it elsewhere?

Reply:
@joshdover I think this separation just proved to be useful with the [concerns](https://github.com/elastic/elasticsearch/pull/96083#pullrequestreview-1428408865) raised by @felixbarny - it will now allow us to remove it from the [default pipeline](https://github.com/elastic/elasticsearch/blob/29ecff1802ca31fcb210bd54164723eeab94b646/x-pack/plugin/core/src/main/resources/logs-default-pipeline.json#L13), but still allow users to easily opt in by calling it from the `logs@custom` pipeline

Reply:
Demonstrated now [in this PR's test](https://github.com/elastic/elasticsearch/pull/96083/commits/3b91268de3ebeea20b31108402adbda36f23b2bc#diff-c0825a64da4d630fddd34f7205fe31a15b258795ea39b1f84f04851bde19daccR4-R17)